File: codex_poting/java/src/main/java/com/example/demo/util/PageCalculator.java
package com.example.demo.util;

public final class PageCalculator {
    private PageCalculator() {
    }

    public static int calculateTotalPages(long totalElements, int size) {
        if (totalElements == 0) {
            return 0;
        }
        return (int) ((totalElements + size - 1) / size);
    }

    public static long calculateOffset(int page, int size) {
        return (long) page * size;
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/util/PageCalculator.kt
package com.example.demo.util

object PageCalculator {
    fun calculateTotalPages(totalElements: Long, size: Int): Int {
        if (totalElements == 0L) {
            return 0
        }
        return ((totalElements + size - 1) / size).toInt()
    }

    fun calculateOffset(page: Int, size: Int): Long {
        return page.toLong() * size
    }
}
